yocto-base/01_base_os.robot

```diff
@@ -0,0 +1,51 @@
+*** Settings ***
+Library             Collections
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../../sonoff-rest-api/sonoff-api.robot
+Resource            ../../rtectrl-rest-api/rtectrl.robot
+Resource            ../../variables.robot
+Resource            ../../keywords.robot
+Resource            ../lib/sd-wire.robot
+Resource            ../lib/linux.robot
+
+Suite Setup         Prepare Test Suite
+Suite Teardown      Log Out And Close Connection
+
+
+*** Test Cases ***
+Booting Platform
+    [Documentation]    This test aims to see whether the device can boot.
+    Variable Should Exist    ${DUT_PASSWORD}
+    Sonoff Power Off
+    Sleep    5s
+    Sonoff Power On
+    Serial Root Login Linux    ${DUT_PASSWORD}
+
+Basic Packages
+    [Documentation]    checks whether tar, time and chronyc utilities exist.
+    Get Utility Version    tar
+    Get Utility Version    time
+    Get Utility Version    chronyc
+
+USB
+    [Documentation]    check whether we can see a USB stick that's plugged in.
+    ...    Also checks whether we mount/umount it and write/read.
+    ${output}=    Telnet.Execute Command    lsblk | grep sda | wc -l
+    ${output}=    Get Line    ${output}    0
+    Should Be Equal As Strings    ${output}    2
+    Telnet.Execute Command    mount /dev/sda1 /mnt
```

Comment:
@artur-rs this could work:
```bash
dmesg | grep -F "USB Mass Storage device detected" --context=10 | grep 'sd[^[:space:]]' | tail -1 | sed 's/.*\(sd[^[:space:]]\).*/\1/'
```
but its very ugly/hard to read (and might not necessarily work in 100% of cases, even though i havent been able to think of a case where it wouldn't). Should it be done anyway?

Reply:
@artur-rs it ended up being a command thats a little different, but i think it should work. I also used `Power Cycle On`. Both of these changes are in the [latest commit.](https://github.com/Dasharo/open-source-firmware-validation/pull/34/commits/f3c0314d8b0db2215c2f4a275cd432b623c2cbfc)

Reply:
Let's stick to this solution for now and if problems occur with the regexp, a new issue will be added. Thanks